_posts/2020-09-17-accessing-snowplow-data-in-real-time-aws/index.md

```diff
@@ -0,0 +1,123 @@
+"kinesis:GetShardIterator",
+"kinesis:DescribeStream",
+"kinesis:ListStreams",
+"logs:CreateLogGroup",
+"logs:CreateLogStream",
+"logs:PutLogEvents"
+```
+
+## 3. Create the lambda function
+Create a new Lambda function by clicking `Create Function` and give it the following properties: 
+* Name: lambda_function_payload
+* Runtime: Python 3
+* Permissions: Use existing role, and select the role you made in the previous step
+![Screenshot](img/create_function.png)
+
+## 4. Create the Python script
+Copy this Python script to a file locally. In my case, it's called lambda_function_payload.py and is inside a subdirectory called lambda_function_payload. This subdirectory is important for the next step. 
+
+```python
+import snowplow_analytics_sdk.event_transformer
+import snowplow_analytics_sdk.snowplow_event_transformation_exception
+import json
+import base64
+
+# Decode batch of lambda records from Base64 to TSV
+def decode_records(update):
+    data = []
+    if "Records" in update:
+        for record in update["Records"]:
+            if "kinesis" in record and "data" in record['kinesis']:
+                # decode base64 data in each record and append to the list
+                decoded_data = base64.b64decode(record['kinesis']['data']).decode('utf-8')
+                data.append(decoded_data)
+    return data
+
+# Entry point for the lambda function
+def lambda_handler(event, context):
+    # Uncomment this line to show the raw batch of events
+    # print("Received event: " + json.dumps(event, indent=2)) 
+
+    # Decode received batch of records from Base64 to Snowplow TSV
+    records = decode_records(event)
+
+    for record in records:
+        try:
+            # Transform record from Snowplow TSV to JSON
+            print(snowplow_analytics_sdk.event_transformer.transform(record))
+            print("Succesfully retrieved and transformed event")
+        except snowplow_analytics_sdk.snowplow_event_transformation_exception.SnowplowEventTransformationException as e:
+            for error_message in e.error_messages:
+                print(error_message)
+            continue
+        
+    return
+```
+
+## 5. Package and upload the Python script
+Package a zip file for lambda with the following script. Copy it to a new file and place it in the directory above the python script. Note that this is being run on a subdirectry with the Python app called lambda_function_payload.py. The Python fuction above is written in Python 3 the script below will install the [Snowplow Python Analytics SDK](https://github.com/snowplow/snowplow/wiki/Python-Analytics-SDK-Setup) in the directory so that it is included in the zip file.
+
+```bash
+#!/bin/bash
+cd lambda_function_payload
+pip3 install snowplow_analytics_sdk -t . --upgrade
+chmod -R 755 .
+zip -r ../lambda_function_payload.zip .
+cd ..
+```
+
+Upload the zip file to lambda in the AWS Console. Since the dependency is a couple of mb, we won't be able to see and edit the code in AWS. 
```

Comment:
```suggestion
Upload the zip file to lambda in the AWS Console. Since the dependency is a couple of mb, we won't be able to see and edit the code in AWS.  

```